toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/combo/ComboBoxField.kt

```diff
@@ -0,0 +1,268 @@
+        isEditable = isEditable,
+        label = state.label,
+        placeholder = state.placeholder,
+        singleLine = true,
+        trailingIcon = Icons.Outlined.List,
+        supportingText = {
+            Text(
+                text = state.description,
+                modifier = Modifier.semantics { contentDescription = "description" },
+            )
+        },
+        interactionSource = interactionSource
+    )
+
+    if (showDialog) {
+        ComboBoxDialog(
+            initialValue = value,
+            values = state.codedValues.map { it.code.toString() },
+            label = state.label,
+            description = state.description,
+            isRequired = isRequired,
+            noValueOption = state.showNoValueOption,
+            noValueLabel = state.noValueLabel.ifEmpty { stringResource(R.string.no_value) },
+            onValueChange = {
+                state.onValueChanged(it)
+            }
+        ) {
+            showDialog = false
+        }
+    }
+
+    LaunchedEffect(interactionSource) {
+        interactionSource.interactions.collect {
+            if (it is PressInteraction.Release) {
+                showDialog = true
+            }
+        }
+    }
+}
+
+@Composable
+internal fun ComboBoxDialog(
+    initialValue: String,
+    values: List<String>,
+    label: String,
+    description: String,
+    isRequired: Boolean,
+    noValueOption: FormInputNoValueOption,
+    noValueLabel: String,
+    onValueChange: (String) -> Unit,
+    onDismissRequest: () -> Unit
+) {
+    var searchText by remember { mutableStateOf("") }
+    val codedValues = if (!isRequired) {
+        if (noValueOption == FormInputNoValueOption.Show) {
+            listOf(noValueLabel) + values
+        } else values
+    } else values
+
+    val filteredList by remember {
+        derivedStateOf {
+            codedValues.filter {
+                it.contains(searchText, ignoreCase = true)
+            }
+        }
+    }
+
+    Dialog(
+        onDismissRequest = onDismissRequest,
+        properties = DialogProperties(usePlatformDefaultWidth = false)
+    ) {
+        Scaffold(
+            topBar = {
+                Column(
+                    modifier = Modifier
+                        .padding(start = 15.dp, top = 15.dp, bottom = 10.dp, end = 10.dp)
+                        .fillMaxWidth(),
+                ) {
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        TextField(
+                            value = searchText,
+                            onValueChange = {
+                                searchText = it
+                            },
+                            modifier = Modifier.weight(1f),
+                            placeholder = {
+                                Text(text = "${stringResource(R.string.filter)} $label")
```

Comment:
the resource string itself ought to do the formatting, so that when it is translated the translator can change the placement of the `label` string within the resource string.

```suggestion
stringResource(R.string.filter, label)
```

Reply:
Ah missed that, thanks.